File: go/raindrops/raindrops.go
// Package raindrops contains the solution of the Raindrops task
package raindrops

import "strconv"

// Convert function for PlingPlangPlong task
// If the number is divisible with 3 return with Pling
// If the number is divisible with 5 return with Plang
// If the number is divisible with 7 return with Plong
func Convert(number int) string {
	s := ""
	if number%3 == 0 {
		s += "Pling"
	}

	if number%5 == 0 {
		s += "Plang"
	}

	if number%7 == 0 {
		s += "Plong"
	}

	if s == "" {
		s += strconv.Itoa(number)
	}

	return s
}

File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {

    fun convert(n: Int): String {
        return mapOf(3 to "Pling", 5 to "Plang", 7 to "Plong")
            .filter { n % it.key == 0 }
            .map { it.value }
            .joinToString("")
            .ifEmpty { n.toString() }
    }
}
